dev/com.ibm.websphere.appserver.thirdparty.eclipselink/src/org/eclipse/persistence/expressions/ExpressionOperator.java

```diff
@@ -0,0 +1,3285 @@
+        v.add("");
+        result.printsAs(v);
+        result.bePrefix();
+        result.setNodeClass(ClassConstants.FunctionExpression_Class);
+        v = NonSynchronizedVector.newInstance(2);
+        v.add(".notLike(");
+        v.add(")");
+        result.printsJavaAs(v);
+        return result;
+    }
+
+    /**
+     * INTERNAL:
+     * Create the NOTNULL operator.
+     */
+    public static ExpressionOperator notNull() {
+        ExpressionOperator result = new ExpressionOperator();
+        result.setType(ComparisonOperator);
+        result.setSelector(NotNull);
+        Vector v = org.eclipse.persistence.internal.helper.NonSynchronizedVector.newInstance();
+        v.add("(");
+        v.add(" IS NOT NULL)");
+        result.printsAs(v);
+        result.bePrefix();
+        result.printsJavaAs(".notNull()");
+        result.setNodeClass(ClassConstants.FunctionExpression_Class);
+        return result;
+    }
+
+    /**
+     * INTERNAL:
+     * Create the NOT operator.
+     */
+    public static ExpressionOperator notOperator() {
+        ExpressionOperator result = new ExpressionOperator();
+        result.setSelector(Not);
+        Vector v = org.eclipse.persistence.internal.helper.NonSynchronizedVector.newInstance();
+        v.add("NOT (");
+        v.add(")");
+        result.printsAs(v);
+        result.bePrefix();
+        result.printsJavaAs(".not()");
+        result.setNodeClass(ClassConstants.FunctionExpression_Class);
+        return result;
+    }
+    
+    /**
+     * INTERNAL:
+     * Build operator.
+     */
+    public static ExpressionOperator nullIf() {
+        return simpleTwoArgumentFunction(NullIf, "NULLIF");
+    }
+
+    /**
+     * INTERNAL:
+     * Create the OR operator.
+     */
+    public static ExpressionOperator or() {
+        return simpleLogical(Or, "OR", "or");
+    }
+
+    /**
+     * INTERNAL:
+     * Build operator.
+     */
+    public static ExpressionOperator power() {
+        return simpleTwoArgumentFunction(Power, "POWER");
+    }
+
+    /**
+     * INTERNAL: Print the collection onto the SQL stream.
+     */
+    public void printCollection(Vector items, ExpressionSQLPrinter printer) {
+        /*
+         * If this ExpressionOperator does not support binding, and the platform allows,
+         * then disable binding for the whole query
+         */
+        if (printer.getPlatform().isDynamicSQLRequiredForFunctions() && Boolean.FALSE.equals(isBindingSupported())) {
+            printer.getCall().setUsesBinding(false);
+        }
+
+        int dbStringIndex = 0;
+        if (isPrefix()) {
+            printer.printString(getDatabaseStrings()[0]);
+            dbStringIndex = 1;
+        }
+
+        // If the operator defines specific indices, use those
+        int[] indices = this.argumentIndices;
+        if(indices == null) {
+            indices = new int[items.size()];
+            for (int i = 0; i < indices.length; i++){
+                indices[i] = i;
+            }
```

Comment:
After initializing the array, should this line be added?
`this.argumentIndices = indices;`
In this way, any subsequent users of this class would not find a null and need to re-create the array, and it would be consistent with the earlier behavior.